lib/fastlane/plugin/browserstack/actions/upload_to_browserstack_app_automate_action.rb

```diff
@@ -0,0 +1,105 @@
+require 'fastlane/action'
+require_relative '../helper/browserstack_helper'
+require 'json'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      BROWSERSTACK_APP_ID ||= :BROWSERSTACK_APP_ID
+    end
+    class UploadToBrowserstackAppAutomateAction < Action
+      SUPPORTED_FILE_EXTENSIONS = ["apk", "ipa", "zip"]
```

Comment:
Supported are just apk and ipa for the below endpoint.

If this plugin has to support espresso and xcui (and earlgrey in the future) as well, then we need to support zip. In that case, the api endpoints will change based on the framework.

Reply:
What is the general endpoint for uploading all kinds of app for testing?

Reply:
there isn't 1 endpoint.
ipa/apk apps: https://api-cloud.browserstack.com/app-automate/upload
apk (espresso test suite): https://api-cloud.browserstack.com/app-automate/espresso/test-suite
zip (xcui test suite): https://api-cloud.browserstack.com/app-automate/xcuitest/test-suite
zip (earlgrey test suite): https://api-cloud.browserstack.com/app-automate/earlgrey/app-dir

Reply:
Currently, we are going to support apk and ipa file upload only, hence using first endpoint : https://api-cloud.browserstack.com/app-automate/upload .